submissions/the-darkesttt/frogger-task/app.js

```diff
@@ -0,0 +1,323 @@
+const playerSkins = [
+    {
+        id: 'boy',
+        src: 'images/char-boy.png',
+    },
+    {
+        id: 'cat-girl',
+        src: 'images/char-cat-girl.png',
+    },
+    {
+        id: 'horn-girl',
+        src: 'images/char-horn-girl.png',
+    },
+    {
+        id: 'pink-girl',
+        src: 'images/char-pink-girl.png',
+    },
+    {
+        id: 'princess-girl',
+        src: 'images/char-princess-girl.png',
+    },
+];
+
+class Menu {
+    bg;
+    menu;
+    menuTitle;
+    menuDescriprion;
+
+    menuList;
+    menuItem;
+
+    menuScore;
+    menuClose;
+
+    constructor() {
+        this.bg = document.createElement('div');
+        this.bg.classList.add('bg');
+        this.menu = document.createElement('div');
+        this.menu.classList.add('menu');
+        this.menuTitle = document.createElement('h1');
+        this.menuTitle.innerText = 'Game Menu';
+        this.menuDescriprion = document.createElement('p');
+    }
+
+    start() {
+
+        this.menuDescriprion.innerText = 'Choose your character!';
+        this.menuList = document.createElement('ul');
+
+        playerSkins.forEach((item) => {
+            this.menuItem = document.createElement('li');
+    
+            const itemImg = document.createElement('img');
+            const imgSrc = item.src;
+            const imgId = item.id;
+            itemImg.setAttribute('src', imgSrc);
+            itemImg.setAttribute('id', imgId);
+            const imgDesc = document.createElement('p');
+            imgDesc.innerText = imgId;
+    
+            this.menuList.appendChild(this.menuItem);
+            this.menuItem.appendChild(itemImg);
+            this.menuItem.appendChild(imgDesc);
+            
+        });
+
+        document.body.appendChild(this.bg);
+        document.body.appendChild(this.menu);
+
+        this.menu.appendChild(this.menuTitle);
+        this.menu.appendChild(this.menuDescriprion);
+        this.menu.appendChild(this.menuList);
+
+        this.menuList.addEventListener('click', (event) => {
+            const listTarget = event.target;
+            if (listTarget.tagName == "UL") return;
+        
+            player.sprite = listTarget.getAttribute("src");
```

Comment:
Here it is fine since we are still learning and the skin picker wasn't even part of the main task, but keep this in mind:
_NEVER_ rely on something being in DOM. Here, for example, getting a path to the skin from the `src` property seems like a neat shortcut, but remember that the user can easily open dev tools in the browser, edit the `src` in the element and it would break everything since there is no such path in your files. This exact code piece could be implemented by giving each "skin button" an ID and then, within JS you could connect each ID to some skin path. Actually, you already have an array like that at the start of the file.